### PR TITLE
Subspecs

### PR DESCRIPTION
**Pull request by alfiehanssen**
This PR sets up a number of subspecs within VIMNetworking. Specifically so that we can use some of our existing model objects inside VIMUpload.

### CODE REVIEW

**Comment by alfiehanssen** [comment]
@huebnerob @ghking, this is ready for review

File: VIMNetworking/Model/VIMQuota.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "VIMModelObject.h"
+
+@interface VIMQuota : VIMModelObject
+
+@property (nonatomic, strong, nullable) NSNumber *hd;
```
Comment (huebnerob) [inline VIMNetworking/Model/VIMQuota.h]: feels like these vars should be more descriptively named, as we've done with other terse var names from the api

File: VIMNetworking/Model/VIMUploadQuota.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+#import "VIMModelObject.h"
+
+@class VIMQuota;
+@class VIMSpace;
+
+@interface VIMUploadQuota : VIMModelObject
+
+@property (nonatomic, strong, nullable) VIMQuota* quota;
+@property (nonatomic, strong, nullable) VIMSpace* space;
```
Comment (huebnerob) [inline VIMNetworking/Model/VIMUploadQuota.h]: `VIMQuantityQuota` and `VIMSizeQuota`? a nit perhaps, but these feel more nonintuitive than they should be

File: VIMNetworking/Model/VIMVideo.m [hunk 3]
```diff
@@ -415,110 +407,4 @@ - (NSInteger)commentsCount
     return (self.canViewComments ? commentsConnection.total.intValue : 0);
 }
 
-#pragma mark - File Selection
-
-- (VIMVideoFile *)hlsFileForScreenSize:(CGSize)size
```
Comment (huebnerob) [inline VIMNetworking/Model/VIMVideo.m]: noice moves

File: VIMNetworking/Model/VIMVideoUtils.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
+//
```
Comment (huebnerob) [inline VIMNetworking/Model/VIMVideoUtils.h]: Should move VIMVideoUtils out of the model directory probably

**Comment by huebnerob** [comment]
dope'ade :lemon:

File: VIMNetworking/Model/VIMVideoUtils.h (hunk 4 above)
Comment (alfiehanssen) [inline VIMNetworking/Model/VIMVideoUtils.h]: They're so tightly coupled with the models though. Would this mean that the Model subspec contains the Utils/ directory?

File: VIMNetworking/Model/VIMQuota.h (hunk 1 above)
Comment (alfiehanssen) [inline VIMNetworking/Model/VIMQuota.h]: Will do

File: VIMNetworking/Model/VIMUploadQuota.h (hunk 2 above)
Comment (alfiehanssen) [inline VIMNetworking/Model/VIMUploadQuota.h]: Will do

**Pull request merged**